src/route-loader.js

```diff
@@ -46,12 +62,22 @@ export class TemplatingRouteLoader extends RouteLoader {
       return childContainer.get(Router);
     };
 
+    if (isPromise) {
+      return config.moduleId.then(viewModelClass => {
+        if (typeof viewModelClass !== 'function') {
+          throw new Error(`Unsupported moduleId config. Expected class / function, actual: ${typeof viewModelClass}`);
+        }
+        instruction.viewModel = { constructor: viewModelClass };
+        return this.compositionEngine.ensureViewModel(instruction);
+      });
+    }
+
     return this.compositionEngine.ensureViewModel(instruction);
   }
 }
 
 function createDynamicClass(moduleId) {
-  let name = /([^\/^\?]+)\.html/i.exec(moduleId)[1];
+  let name = /([^\/^\?]+)\.html$/i.exec(moduleId)[1];
```

Comment:
reasonable change, was this necessary for your PR or just kinda snuck in there?

Reply:
I'm pretty sure I tried to delete this. Can remove this change before we finalize this though